### DISCUSSION TITLE
Can't fork on mobile phone.

### DISCUSSION THREAD

**Discussion by mr-cheffy** [community/community#10964] [Other Feature Feedback, Questions, & Ideas]
Hi, I wanted to fork a repository only to realize that the fork button disappeared. It would be cool if you added it. What I did what to was to enable the "display as desktop" on my browser and I forked it from there.

**Comment by amrehman** [comment]
Hi @mauro-balades , thanks for getting in touch.
Just to understand your workflow, can you please confirm for me if you are using the web page on the mobile or are you using one of the client apps on iOS/Android.

JFYI, forking is not supported on our client apps, but we will start this discussion amongst our team regarding this feature. 

Thanks a lot for your feedback.

**Reply by mr-cheffy** [reply]
I was using the web page

**Reply by elidoran** [reply]
I can’t get the Fork button to show in any mobile browser.

My desired workflow is:
1. Visit someone else’s repo via web browser or GitHub iOS app
2. Login to GitHub (if not already)
3. Click Fork to create my own fork of the repo
4. Switch to my fork of the repo
5. Create a tag/release of the fork
6. Go to the releases/tags to download the project zip/tgz file
7. Transfer that file from my iPhone to iMac via direct cable
8. Use open source project on my iMac 

I hear you, why not download on my iMac ? I don’t have “home internet”. I use my phone’s data. 
For a repo with its own release/tag I can at least get a link to that, but, it’s not the most current, so, if I wanted to grab the newest code… 
It seems like there should be a standard “download this repo” link in the browser version but I don’t see one. I think there is in a desktop sized browser view, so, …
Anyway, for those of us relying on mobile only access some changes would nice, including the missing fork button (both browser and mobile app).

JFYI, I’m looking to download repo munificent/craftinginterpreters to play around with language creation.

**Reply by A5H1Q** [reply]
> I can’t get the Fork button to show in any mobile browser.

why not try using desktop mode, if you're in chrome android

**Reply by peter-avila** [reply]
Thanks for looking into adding this feature. It used to be very convenient to be able to fork a project on the go, but that no longer works on iOS browsers. Not having to switch to desktop mode would be even better.

**Reply by longjiang** [reply]
I’m just curious why is that button hidden on small screens, what’s the rationale behind hiding it?

**Reply by Apachirx** [reply]
> Hi @mauro-balades , thanks for getting in touch.
> 
> Just to understand your workflow, can you please confirm for me if you are using the web page on the mobile or are you using one of the client apps on iOS/Android.
> 
> 
> 
> JFYI, forking is not supported on our client apps, but we will start this discussion amongst our team regarding this feature. 
> 
> 
> 
> Thanks a lot for your feedback.

**Comment by ClaraBara22** [comment]
I have GitHub on my computer desktop. Can you switch to that?

**Reply by elidoran** [reply]
Hello ClaraBara22. 
Please note I wrote “I hear you, why not download on my iMac ? I don’t have ‘home internet’. I use my phone’s data”.
So, the answer to your question is: my desktop, and all my other devices, do not have internet access. I only have internet via my phone. It cannot hotspot to other devices to share that internet.
And, the only reason I can’t download a repo is because GitHub’s UI design prevents it in both their web pages and their iOS app. Their web design uses a fluid design which changes based on the view size. The small “mobile” size doesn’t show the usual download links or the fork button. It should, but some designers there decided they didn’t want to allow that on mobile, which is a design flaw. 
A friend of mine helped me out with this using her iPad. No desktop required. Simply the larger view size shows the needed links in the page so they can be used to download the repo or fork it.
So, the only reason mobile users can’t do the download feature, and the fork, is because their web page’s fluid UI is designed to not show those UI components on small screens, and their app doesn’t support downloading a branch or forking a repo. 
It seems an easy fix is to tweak the small view UI to have those buttons. And, of course, add them to the app’s UI. It has a download for releases/tags; it needs a download for branches, and it would be nice for commits, too.

**Comment by ClaraBara22** [comment]
I have desktop too

**Comment by Sodykhorn** [comment]
Thank you

**Comment by vanpham981986** [comment]
display as desktop

**Reply by peter-avila** [reply]
That used to work, but unfortunately no longer does on iOS.

**Reply by byronvaz47** [reply]
does not work it’s RIDICULOUS

**Comment by lukeswitz** [comment]
@github-staff Can we get an update as to why you’d cut features from mobile? Pushing us to desktop only will fail for you.

**Reply by RNahapeti** [reply]
@amrehman @github-staff I'm noticing the same issue on my laptop (macOS) - anytime I have my screen split and resolution for my browser falls under tablet view or smaller, the "Code" and "Fork" buttons disappear. I'm forced to full screen the site to be able to access either of those options.

Why exactly are you hiding buttons at different media resolutions? What benefit could you possibly be gaining by removing the two most important buttons on a repo page at lower resolutions?

**Reply by dreampiggy** [reply]
Agreed. I don't think this decision will increase GitHub UV on mobile. You're doing something has no data benefits on your job.

**Comment by Sodykhorn** [comment]
Thank you

ទទួល Outlook សម្រាប់ iOS<https://aka.ms/o0ukef>
________________________________

**Comment by brundabharadwaj** [comment]
> Hi, I wanted to fork a repository only to realize that the fork button disappeared. It would be cool if you added it. What I did what to was to enable the "display as desktop" on my browser and I forked it from there.

——-
Could not get this option on ios!

**Comment by minsley** [comment]
I managed on mobile safari by requesting desktop site and zooming to 50%

**Reply by dreampiggy** [reply]
The only way on iPhone(not iPad) device to fork repo.

Testing Date : 2022-12-09

**Reply by YakShavingCatHerder** [reply]
> The only way on iPhone(not iPad) device to fork repo.
> 
> 
> 
> Testing Date : 2022-12-09

Can confirm. Firefox and Brave on iPhone, even when requesting desktop site, only allow "Watch" and "Star". 

At least Micro$h1t is consistent in seemingly going out of their way to ruin the smallest things for absolutely no reason.

**Comment by tvquizphd** [comment]
One solution is to manually visit `https://github.com/owner/repo/fork`
where `https://github.com/owner/repo`
is the normal path of the repository you would like to fork.

**Reply by mr-cheffy** [reply]
Yes, but that's annoying on a mobile phone.

**Reply by mschauer** [reply]
This seems to have been disabled too which feels malicious

**Comment by TonyGravagno** [comment]
I frequently use my Android phone with the Firefox browser (like right now) to browse GitHub for projects. I've also been frustrated that a common Fork button isn't available here. Makes no sense to me since there's no client side activity.

I also support adding the Fork button to the small form factor browser display, either in the main repo pages or on the page with the list of forks. A simple a-type link to the /fork page will do the trick if a button costs too much. ;)

Thanks

**Reply by Kojonhyiraa** [reply]
I still can't see my fork button

**Comment by mr-cheffy** [comment]
It looks like they added the button now, thanks everyone for ur answers

**Reply by shilezi** [reply]
Where?

**Reply by mr-cheffy** [reply]
![Screenshot_2023-06-20-07-14-58-212_org mozilla firefox-edit](https://github.com/community/community/assets/91018726/ffbc6929-c00e-4f00-80eb-5b50b35018d1)

**Reply by alin256** [reply]
I still do not see button. Did they forget to update the iPad app?
... and I do not see it on my iPhone app either...
![image](https://github.com/community/community/assets/7563037/cc472bbf-e9ff-4cd0-aa56-ad5238ef2978)

**Reply by mr-cheffy** [reply]
They didn't add it to the app, just mobile web version

**Reply by alin256** [reply]
Do you know if I can vote for an issue to propagate the improvement to the app?

**Reply by mr-cheffy** [reply]
no, sry

**Reply by DarkError09** [reply]
How to run any GitHub code on GitHub web in mobile help

**Comment by erdoukki** [comment]
still no more fork button, no more...
not real tweak...
again a small hack making apk useless...
wtf !

why coder need to get more bad tools ?
apple has stolen your fork feature ?

juste make another false ordinary real kill...
oups, fork...

allo @bill, what's the pbm with you little boy ?
@github team, apk tool, juste need to learn coding ?
or any real reason ?
2024 need to be bad than 99 ??
really ?
@bill, call mama again, may be big blue as more dos to sell (to) you ?

happy new reboot of the time loop...

ag
